drivers/led/led_gpio.c

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT gpio_leds
+
+/**
+ * @file
+ * @brief GPIO driven LEDs
+ */
+
+#include <drivers/led.h>
+#include <drivers/gpio.h>
+#include <device.h>
+#include <zephyr.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(led_gpio, CONFIG_LED_LOG_LEVEL);
+
+#define DEV_CFG(dev)	((const struct led_gpio_config *) ((dev)->config))
+
+struct led_gpio {
+	const struct device *dev;
+	gpio_pin_t pin;
+	gpio_flags_t flags;
+};
+
+struct led_gpio_config {
+	size_t num_leds;
+	const struct led_gpio *led;
+};
+
+static int led_gpio_blink(const struct device *dev, uint32_t led,
+			  uint32_t delay_on, uint32_t delay_off)
+{
+	return -ENOTSUP;
```

Comment:
I know we don't need it but maybe we could add blink using workqueue?

Reply:
btw - if function is not supported won't it be enough to not add it to ops

Reply:
I prefer to keep the driver as simple and small as possible (it could be used by memory-limited devices).
Work in driver would need to be added for LED every color channel. Application could use a single work for whole multicolor LED. Adding blinking to an application would be simple too.